ksqldb-functional-tests/src/test/resources/query-validation-tests/row-offset-and-partition.json

```
@@ -119,6 +150,622 @@
+      ],
+      "inputs": [
+        {"topic":  "test_topic", "key": 1, "value": {"foo": "a"}, "timestamp": 5},
+        {"topic":  "test_topic", "key": 2, "value": {"foo": "b"}, "timestamp": 6},
+        {"topic":  "test_topic", "key": 3, "value": {"foo": "c"}, "timestamp":  7}
+      ],
+      "outputs": [
+        {"topic":  "OUTPUT", "key": 1, "value": {"RP": 3, "RO": 0}, "timestamp": 5},
+        {"topic":  "OUTPUT", "key": 2, "value": {"RP": 3, "RO": 3}, "timestamp": 6},
+        {"topic":  "OUTPUT", "key": 3, "value": {"RP": 3, "RO": 6}, "timestamp": 7}
+      ],
+      "post": {
+        "sources": [
+          {"name": "OUTPUT", "type": "stream", "schema": "ID INT KEY, RP INT, RO BIGINT"}
+        ]
+      }
+    },
+    {
+      "name": "PARTITION BY ROWOFFSET",
+      "properties": {
+        "ksql.rowpartition.rowoffset.enabled": true
+      },
+      "statements": [
+        "CREATE STREAM input (id INT KEY, val INT) WITH (kafka_topic='input', value_format='JSON');",
+        "CREATE STREAM output AS select * FROM input PARTITION BY ROWOFFSET;"
+      ],
+      "inputs": [
+        {"topic": "input", "key": 10, "value": {"val":  5}},
+        {"topic": "input", "key": 9, "value": {"val":  6}},
+        {"topic": "input", "key": 8, "value": {"val":  7}}
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "key": 0, "value": {"ID": 10, "VAL":  5}},
+        {"topic": "OUTPUT", "key": 1, "value": {"ID": 9, "VAL":  6}},
+        {"topic": "OUTPUT", "key": 2, "value": {"ID": 8, "VAL":  7}}
+      ],
+      "post": {
+        "topics": {
+          "blacklist": ".*-repartition"
+        },
+        "sources": [
+          {"name": "OUTPUT", "type": "stream", "schema": "ROWOFFSET BIGINT KEY, val INT, id INT"}
+        ]
+      }
+    },
+    {
+      "name": "PARTITION BY ROWPARTITION",
+      "comments": "Not the most useful PARTITION BY!",
+      "properties": {
+        "ksql.rowpartition.rowoffset.enabled": true
+      },
+      "statements": [
+        "CREATE STREAM input (id INT KEY, val INT) WITH (kafka_topic='input', value_format='JSON');",
+        "CREATE STREAM output AS select * FROM input PARTITION BY ROWPARTITION;"
+      ],
+      "inputs": [
+        {"topic": "input", "key": 10, "value": {"val":  5}},
+        {"topic": "input", "key": 9, "value": {"val":  6}},
+        {"topic": "input", "key": 8, "value": {"val":  7}}
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "key": 0, "value": {"ID": 10, "VAL":  5}},
+        {"topic": "OUTPUT", "key": 0, "value": {"ID": 9, "VAL":  6}},
+        {"topic": "OUTPUT", "key": 0, "value": {"ID": 8, "VAL":  7}}
+      ],
+      "post": {
+        "topics": {
+          "blacklist": ".*-repartition"
+        },
+        "sources": [
+          {"name": "OUTPUT", "type": "stream", "schema": "ROWPARTITION INT KEY, val INT, id INT"}
+        ]
+      }
+    },
+    {
+      "name": "Try to select ROWPARTITION in persistent query without aliases",
```

```suggestion
      "name": "Should fail if select ROWPARTITION without aliases",
```
and similarly below.